paper.md

```diff
@@ -0,0 +1,140 @@
+the typical CMake workflow encapsulates running the build system, the
+present manuscript ignores the "generator" distinction). Therefore with the
+advent of the exascale era, there is a desperate need for a robust CMake
+development ecosystem.
+
+CMake already contains a number of tools and features that facilitate
+development of the target software package. For example, CMake's
+`find_package` module simplifies dependency management and the `CTest` package
+eases the process of testing the resulting software. Additional CMake tools
+can be created by writing CMake modules. While the CMake language is
+flexible and relatively simple, it is not without its pitfalls. Unfortunately
+tools  to facilitate the development of CMake modules are relatively sparse.
+Here we introduce CMinx, a tool for generating API documentation for CMake
+modules. API documentation is arguably one of the most basic elements of a
+software development ecosystem and it is our hope that CMinx will serve as the
+foundation for a robust CMake development ecosystem.
+
+Anecdotal evidence [@official_solution] indicates that the developers of the
+CMake language (*i.e.*, Kitware) internally write their API documentation
+using reST and Sphinx. Following best practices, this reST API documentation
+resides next to the CMake code being described. Kitware has also written a
+Sphinx plugin that makes it easy to extract the API documentation as part of a
+normal Sphinx workflow. This Sphinx plugin is distributed with the source code
+for the CMake interpreter, and is also available in a GitHub repository mirror
+[@sphinx_plugin]. For completeness, we note that similar Sphinx plugins
+[@official_sphinx_domain; @marco_koch] have been independently developed, but
+appear to now be abandoned.
+
+To our knowledge, all of the aforementioned Sphinx plugins simply extract the
+reST API documentation verbatim. Notably this means the developer is responsible
+for manually: adding the function/macro signatures to the documentation, listing
+the function's parameters, and the overall formatting (*e.g.*, titles and
+subtitles, creation of parameter tables, underline/overline consistency). For
+build system developers maintaining one or two CMake modules, these are
+admittedly pretty minor inconveniences; however, for organizations
+maintaining a substantial CMake code base, these "minor inconveniences" can
+impact productivity, particularly when ensuring consistency. CMinx
+differs from previous solutions primarily in three ways. First, CMinx
+understands the grammar of the CMake language, meaning CMinx can automatically
+generate some of the documentation by "reading" the source code. Second, CMinx
+generates static reST files; this decreases the number of stub files developers
+need to maintain and makes it easier for the resulting documentation to be
+used in workflows which do not rely on Sphinx. Finally, CMinx has a CMake API
```

Comment:
> Finally, CMinx has a CMake API
so that it integrates more easily into existing CMake workflows.

I'm not quite sure why but something doesn't seem quite right about this sentence. Maybe something like "Finally, CMinx has a CMake API to integrate more easily with existing CMake workflows"?

Or perhaps something like "Finally, CMinx has a CMake API to allow easy integration with existing CMake workflows"